src/utils/callHelpers.ts

```diff
@@ -1,12 +1,49 @@
 import BigNumber from 'bignumber.js'
 import { DEFAULT_GAS_LIMIT, DEFAULT_TOKEN_DECIMAL } from 'config'
-import { ethers } from 'ethers'
+import { ethers, Contract } from 'ethers'
 import { BIG_TEN, BIG_ZERO } from './bigNumber'
 
 const options = {
   gasLimit: DEFAULT_GAS_LIMIT,
 }
 
+/**
+ * Estimate the gas needed to call a function, and add a 10% margin
+ * @param contract Used to perform the call
+ * @param methodName The name of the methode called
+ * @param args An array of arguments to pass to the method
+ * @returns https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt
+ */
+export const estimageGas = async (contract: Contract, methodName: string, methodArgs: any[]) => {
+  if (!contract[methodName]) {
+    throw new Error(`Method ${methodName} doesn't exist on ${contract.address}`)
+  }
+  const rawGasEstimation = await contract.estimateGas[methodName](...methodArgs)
+  const gasEstimation = rawGasEstimation
+    .mul(ethers.BigNumber.from(10000).add(ethers.BigNumber.from(1000)))
```

Comment:
What does this do? 11,000? Why not use a constant?

Reply:
Alternatively, can we use `parseUnit`?

Reply:
https://github.com/pancakeswap/pancake-swap/blob/c48c7861a32bc486184d80e52c2ab22d7d779710/src/utils/index.ts#L54